src/pages/product-details/product-details-page.js

```diff
@@ -0,0 +1,250 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { Card } from '@material-ui/core';
+import { useStyles } from './product-details-page.styles';
+
+import Comments from './comments';
+import SimilarProducts from './similar-products';
+import ProductSizes from './product-sizes';
+import ProductSubmit from './product-submit';
+import ProductFeatures from './product-features/product-features';
+import ProductInfo from './product-info';
+import ProductImages from './product-images';
+import { Loader } from '../../components/loader/loader';
+import {
+  getFiltredProducts,
+  getProduct,
+  setCategoryFilter
+} from '../../redux/products/products.actions';
+
+import { DEFAULT_SIZE } from '../../configs';
+
+const ProductDetails = ({ match }) => {
+  const { id } = match.params;
+  const { product, isLoading, productUrl, categoryFilter } = useSelector(
+    ({ Products, router }) => ({
+      categoryFilter: Products.filters.categoryFilter,
+      product: Products.product,
+      isLoading: Products.productLoading,
+      productUrl: router.location.pathname
```

Comment:
Why do you need `productUrl` at all?

Reply:
I need it for wishlist and cart, so the user can redirect from them to page with product